frontend/public/components/monitoring/query-browser.tsx

```diff
@@ -223,59 +225,78 @@ const graphContainer = (
-    minY = 0;
-  } else if (minY < 0 && maxY < 0) {
-    maxY = 0;
-  }
+    const domain = { x: xDomain || [Date.now() - span, Date.now()], y: undefined };
+
+    let yTickFormat = formatValue;
+
+    if (!isStack) {
+      // Set a reasonable Y-axis range based on the min and max values in the data
+      const findMin = (series) => _.minBy(series, 'y');
+      const findMax = (series) => _.maxBy(series, 'y');
+      let minY = _.get(findMin(_.map(data, findMin)), 'y', 0);
+      let maxY = _.get(findMax(_.map(data, findMax)), 'y', 0);
+      if (minY === 0 && maxY === 0) {
+        minY = -1;
+        maxY = 1;
+      } else if (minY > 0 && maxY > 0) {
+        minY = 0;
+      } else if (minY < 0 && maxY < 0) {
+        maxY = 0;
+      }
 
-  const xTickFormat = span < 5 * 60 * 1000 ? twentyFourHourTimeWithSeconds : twentyFourHourTime;
+      domain.y = [minY, maxY];
 
-  const yTickFormat =
-    Math.abs(maxY - minY) < 0.005 ? (v) => (v === 0 ? '0' : v.toExponential(1)) : formatValue;
+      if (Math.abs(maxY - minY) < 0.005) {
+        yTickFormat = (v) => (v === 0 ? '0' : v.toExponential(1));
+      }
+    }
 
-  return (
-    <div ref={containerRef} style={{ width: '100%' }}>
-      {width > 0 && (
-        <Chart
-          containerComponent={graphContainer}
-          domain={{ x: xDomain || [Date.now() - span, Date.now()], y: [minY, maxY] }}
-          domainPadding={{ y: 1 }}
-          height={200}
-          scale={{ x: 'time', y: 'linear' }}
-          theme={chartTheme}
-          width={width}
-        >
-          <ChartAxis tickCount={5} tickFormat={xTickFormat} />
-          <ChartAxis crossAxis={false} dependentAxis tickCount={6} tickFormat={yTickFormat} />
-          <ChartGroup>
-            {_.map(data, (values, i) => (
-              <ChartLine key={i} data={values} />
-            ))}
-          </ChartGroup>
-        </Chart>
-      )}
-    </div>
-  );
-});
+    const xTickFormat = span < 5 * 60 * 1000 ? twentyFourHourTimeWithSeconds : twentyFourHourTime;
+
+    return (
+      <div ref={containerRef} style={{ width: '100%' }}>
+        {width > 0 && (
+          <Chart
+            containerComponent={graphContainer}
+            domain={domain}
+            domainPadding={{ y: 1 }}
+            height={200}
+            scale={{ x: 'time', y: 'linear' }}
+            theme={chartTheme}
+            width={width}
+          >
+            <ChartAxis tickCount={5} tickFormat={xTickFormat} />
+            <ChartAxis crossAxis={false} dependentAxis tickCount={6} tickFormat={yTickFormat} />
+            {isStack ? (
+              <ChartStack>
+                {_.map(data, (values, i) => (
+                  <ChartArea key={i} data={values} />
+                ))}
+              </ChartStack>
+            ) : (
+              <ChartGroup>
+                {_.map(data, (values, i) => (
+                  <ChartLine key={i} data={values} />
+                ))}
+              </ChartGroup>
+            )}
```

Comment:
We could probably just conditionally set something like a ChartGroupComponent variable based on the value of  `isStack`:

```tsx
const ChartGroupComponent = isStack ? ChartStack : ChartGroup;
// ...
<ChartGroupComponent>
  {_.map(data, (values, i) => (
    <ChartArea key={i} data={values} />
  ))}
</ChartGroupComponent>
```

Reply:
We would also need to select either `ChartArea` or `ChartLine` depending on `isStack`, so I'm not sure it ends up any simpler.

```
const ChartGroupComponent = isStack ? ChartStack : ChartGroup;
const ChartLineComponent = isStack ? ChartArea : ChartLine;
// ...
<ChartGroupComponent>
  {_.map(data, (values, i) => (
    <ChartLineComponent key={i} data={values} />
  ))}
</ChartGroupComponent>
```